libsolidity/analysis/ControlFlowBuilder.cpp

```diff
@@ -25,7 +25,40 @@ using namespace solidity::langutil;
 using namespace solidity::frontend;
 using namespace std;
 
-ControlFlowBuilder::ControlFlowBuilder(CFG::NodeContainer& _nodeContainer, FunctionFlow const& _functionFlow):
+namespace {
+
+bool hasExit(FunctionFlow const& _functionFlow)
+{
+	// Nodes to traverse
+	set<CFGNode const*> nodesToTraverse{_functionFlow.entry};
+	// Nodes traversed (to prevent loops)
+	set<CFGNode const*> nodesTraveresed{};
+
+	while (!nodesToTraverse.empty())
+	{
+		CFGNode const* currentNode = *nodesToTraverse.begin();
+		nodesToTraverse.erase(nodesToTraverse.begin());
+		nodesTraveresed.emplace(currentNode);
+
+		for (auto const& exit: currentNode->exits)
+			if (exit == _functionFlow.exit)
+				return true;
+			else if (!nodesTraveresed.count(exit))
+				nodesToTraverse.emplace(exit);
+	}
+	return false;
```

Comment:
Do you think we can replace this by `BreadthFirstSearch` from https://github.com/ethereum/solidity/blob/80071d59c54a65407d6a2aaa50e355224a8f0ca6/libsolutil/Algorithms.h#L96